nasa/harmony
# Harmony 1538 2

## Jira Issue ID
HARMONY-1538

## Description
Moves services (including harmony) into the `services` directory and the `util` library under the `packages` directory. This is in preparation for using monrepo workspaces which, among other benefits, should make enforcing separation of responsibilities easer. This is a "big" change that doesn't actually change much code (mostly just imports).

You should be able to develop as usual - the top level `npm` scripts still work and I have added some (like `push-image-all` to push imagers to ECR) that should make things easier.

## Local Test Steps

1. Check out this branch
2. Run `npm install`
3. Build all the images for local development using `npm run build-all` optionally followed by `npm run build` (only needed if you want to run harmon in k8s).
4. Start up harmony locally as usual with `bin/boot-strap-harmony` and optionally `bin/start-dev-services`.
5. Run queries

For Sandbox Deployments
6. `export VERSION=<some version>
7. Build X86 version of the images using `npm run build-all-m1`
8. Push the images to ECR
```
export AWS_PROFILE=harmony-sandbox
npm run push-image-all
```
9. Check out branch `harmony-1538-2` of the harmony-ci-cd repo
10. Deploy to sandbox as usual
11. Run queries

## PR Acceptance Checklist
